packages/eslint-plugin/src/rules/no-bad-reference.ts

```diff
@@ -1,58 +1,201 @@
-import { TSESTree } from "@typescript-eslint/utils";
-import { commentsMatching, createRule } from "../util";
+import { ESLintUtils, TSESTree } from "@typescript-eslint/utils";
+import { createRule, findTypesPackage } from "../util";
+import * as ts from "typescript";
+import path from "path";
+import { isDeclarationPath } from "@definitelytyped/utils";
 
-type MessageId = "referencePathPackage" | "referencePathTest" | "referencePathOldVersion";
 const rule = createRule({
   name: "no-bad-reference",
   defaultOptions: [],
   meta: {
     type: "problem",
     docs: {
-      description: `Forbids <reference path="./vNN"/> in all files, <reference path="../etc"/> in declaration files, and all <reference path> in test files.`,
+      description:
+        "Forbids bad references, including those that resolve outside of the package or path references in non-declaration files.",
       recommended: "error",
     },
     messages: {
-      referencePathPackage:
-        "Don't use <reference path> to reference another package. Use an import or <reference types> instead.",
-      referencePathTest:
-        "Don't use <reference path> in test files. Use <reference types> or include the file in 'tsconfig.json'.",
-      referencePathOldVersion: "Don't use <reference path> to reference an old version of the current package.",
+      importOutside:
+        'The import "{{text}}" resolves outside of the package. Use a bare import to reference other packages.',
+      importLeaves: 'The import "{{text}}" resolves to the current package, but uses relative paths.',
+      referenceOutside:
+        'The reference "{{text}}" resolves outside of the package. Use a global reference to reference other packages.',
+      referenceLeaves: 'The reference "{{text}}" resolves to the current package, but uses relative paths.',
+      testReference:
+        'The path reference "{{text}}" is disallowed outside declaration files. Use "<reference types>" or include the file in tsconfig instead.',
```

Comment:
Wouldn't that be the wrong path? Like, instead of writing `../blah`, they should write `my-package/bar`, not just the same string in a different reference kind?

Reply:
you are right. not sure what the best way to express this is then.